pandas/_libs/groupby.pyx

```diff
@@ -480,6 +480,12 @@ ctypedef fused add_t:
     complex128_t
     object
 
+ctypedef fused common_t:
+    float64_t
+    float32_t
+    complex64_t
+    complex128_t
```

Comment:
Would be nice to avoid adding more new fused types, xref #43774. This almost matches `add_t`, except for object - see #43467 for some further discussion on sharing between `sum` and `mean`.

Not sure what's best to do here - the patch as is makes sense, but would be nice if `complex` could be handled through `group_add` to avoid adding more complexity here

Reply:
long-term i agree with @mzeitlin11 but for now I think this is fine.  one nitpick would be to ask for a name more descriptive than `common`

Reply:
is `mean_t` better?